rfc/0054-sdk3-management-apis.md

```diff
@@ -582,6 +666,22 @@ void BuildDeferredIndexes(string bucketName, [options])
 
   * `Timeout` or `timeoutMillis` (`int`/`duration`) - the time allowed for the operation to be terminated. This is controlled by the client.
 
+  * `CollectionName` - the name of the collection to restrict indexes to.
+  * `ScopeName` - the name of the scope to restrict indexes to.
+    * If either `CollectionName` or `ScopeName` are set then both *must* be set.
+
+### N1QL
+
+No collection set:
+```
+"BUILD INDEX ON `bucketName` (`index1`, `index2`)"
+```
+
+Collection set:
+```
+"BUILD INDEX ON `bucketName`.`scopeName`.`collectionName` (`index1`, `index2`)"
```

Comment:
these two statements are not correct. Look, the method called `BuildDeferredIndexes` and described as "builds all indexes which are currently in deferred state".

Do we need scope/collection here at all?

The correct statement to build all deferred indexes associated with  current bucket will be

```sql
BUILD INDEX ON `{bucket_name}` (
  (
    SELECT RAW name FROM system:indexes 
    WHERE ((keyspace_id = "{bucket_name}" AND bucket_id IS MISSING) 
        OR (bucket_id = "{bucket_name}")) 
        AND state = "deferred"
  )
)
```

And if we still want to restrict this method with scope and collection, then the second statement should be

```sql
BUILD INDEX ON `{bucket_name}`.`{scope_name}`.`{collection_name}` (
  (
    SELECT RAW name FROM system:indexes 
    WHERE bucket_id = "{bucket_name}"
      AND scope_id = "{scope_name}"
      AND keyspace_id = "{collection_name}"
      AND state = "deferred"
  )
)
```